modules/dxp/apps/commerce/commerce-machine-learning-test/src/testIntegration/java/com/liferay/commerce/machine/learning/forecast/alert/internal/upgrade/v1_0_1/test/CommerceMachineLearningForecastPortletPreferencesUpgradeProcessTest.java

```diff
@@ -0,0 +1,136 @@
+		new LiferayIntegrationTestRule();
+
+	@Test
+	public void testUpdateCommerceMachineLearningForecastPortletPreferencesExternalReferenceCode()
+		throws Exception {
+
+		Group group = GroupTestUtil.addGroup();
+
+		ServiceContext serviceContext =
+			ServiceContextTestUtil.getServiceContext(group.getGroupId());
+
+		AssetVocabulary assetVocabulary =
+			AssetVocabularyLocalServiceUtil.addVocabulary(
+				serviceContext.getUserId(), group.getGroupId(),
+				RandomTestUtil.randomString(), serviceContext);
+
+		AssetCategory assetCategory = _assetCategoryLocalService.addCategory(
+			serviceContext.getUserId(), group.getGroupId(),
+			RandomTestUtil.randomString(), assetVocabulary.getVocabularyId(),
+			serviceContext);
+
+		Layout layout = LayoutTestUtil.addTypePortletLayout(group);
+
+		String defaultPreferences = StringBundler.concat(
+			"<portlet-preferences><preference><name>assetCategoryIds",
+			"</name><value>", assetCategory.getCategoryId(),
+			"</value></preference></portlet-preferences>");
+
+		PortletPreferences portletPreferences =
+			_portletPreferencesLocalService.addPortletPreferences(
+				group.getCompanyId(), PortletKeys.PREFS_OWNER_ID_DEFAULT,
+				PortletKeys.PREFS_OWNER_TYPE_LAYOUT, layout.getPlid(),
+				CommerceDashboardPortletKeys.COMMERCE_DASHBOARD_FORECASTS_CHART,
+				_portletLocalService.fetchPortletById(
+					group.getCompanyId(),
+					CommerceDashboardPortletKeys.
+						COMMERCE_DASHBOARD_FORECASTS_CHART),
+				defaultPreferences);
+
+		_runUpgrade();
+
+		portletPreferences =
+			_portletPreferencesLocalService.getPortletPreferences(
+				portletPreferences.getPortletPreferencesId());
+
+		javax.portlet.PortletPreferences jxPortletPreferences =
+			PortletPreferenceValueLocalServiceUtil.getPreferences(
+				portletPreferences);
+
+		Assert.assertNotNull(
+			jxPortletPreferences.getValue(
+				"assetCategoryExternalReferenceCodes", null));
+		Assert.assertEquals(
+			assetCategory.getExternalReferenceCode(),
+			GetterUtil.getString(
+				jxPortletPreferences.getValue(
+					"assetCategoryExternalReferenceCodes", null)));
+	}
+
+	private void _runUpgrade() throws Exception {
+		UpgradeProcess upgradeProcess = UpgradeTestUtil.getUpgradeStep(
+			_upgradeStepRegistrator, _CLASS_NAME);
+
+		upgradeProcess.upgrade();
+	}
+
+	private static final String _CLASS_NAME =
+		"com.liferay.commerce.machine.learning.forecast.alert.internal." +
+			"upgrade.v1_0_1." +
+				"CommerceMachineLearningForecastPortletPreferencesUpgradeProcess";
+
+	@Inject
+	private static AssetCategoryLocalService _assetCategoryLocalService;
+
+	@Inject
+	private static PortletLocalService _portletLocalService;
+
+	@Inject
+	private static PortletPreferencesLocalService
+		_portletPreferencesLocalService;
+
+	@Inject(
+		filter = "(&(component.name=com.liferay.commerce.machine.learning.forecast.alert.internal.upgrade.registry.CommerceMachineLearningForecastServiceUpgradeStepRegistrator))"
```

Comment:
`CommerceMachineLearningForecastService` -> `CommerceDashboardWeb`

Pay attention to the package as well when you move it.